src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -8,23 +8,165 @@
+            studentsTable.getRows().add(row);
+            studentIndex++;
+        }
+    }
+    
+    public CourseDetailsStudentsTableRow createStudentsTableRow(int studentIndex, StudentAttributes student) {
+        CourseDetailsStudentsTableRow row = new CourseDetailsStudentsTableRow(student);
+        
+        boolean isDisabled = !currentInstructor.isAllowedForPrivilege(student.section, 
+                                                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS);
+        ElementTag viewButton = createButton("View", "btn btn-default btn-xs", null, getCourseStudentDetailsLink(student),
+                                             Const.Tooltips.COURSE_STUDENT_DETAILS, "tooltip", null, isDisabled);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        ElementTag editButton = createButton("Edit", "btn btn-default btn-xs", null, getCourseStudentEditLink(student),
+                                             Const.Tooltips.COURSE_STUDENT_EDIT, "tooltip", null, isDisabled);
+        
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        ElementTag sendInviteButton = createButton("Send Invite", "btn btn-default btn-xs", null, 
+                                                   getCourseStudentRemindLink(student), Const.Tooltips.COURSE_STUDENT_REMIND, 
+                                                   "tooltip", "return toggleSendRegistrationKey()", isDisabled);
+   
+        isDisabled = !currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        String onClick = "return toggleDeleteStudentConfirmation('" + sanitizeForJs(student.course) 
+                         + "','" + sanitizeForJs(student.name) + "')";
+        ElementTag deleteButton = createButton("Delete", "btn btn-default btn-xs", null, getCourseStudentDeleteLink(student),
+                                               Const.Tooltips.COURSE_STUDENT_DELETE, "tooltip", onClick, isDisabled);
+        
+        ElementTag allRecordsButton = createButton("All Records", "btn btn-default btn-xs", null, 
+                                                   getStudentRecordsLink(student), Const.Tooltips.COURSE_STUDENT_RECORDS, 
+                                                   "tooltip", null, false);
 
-    public InstructorAttributes currentInstructor;
-    public CourseDetailsBundle courseDetails;
-    public List<StudentAttributes> students;
-    public List<InstructorAttributes> instructors;
-    public String studentListHtmlTableAsString;
+        row.getActions().add(viewButton);
+        row.getActions().add(editButton);
+        if (!student.isRegistered()) {
+            row.getActions().add(sendInviteButton);
+        }
+        row.getActions().add(deleteButton);
+        row.getActions().add(allRecordsButton);
```

Comment:
Is it possible to create `List<ElementTag>`s in this method scope, then pass them into `CourseDetailsStudentsTableRow`?
It's odd getting the lists to add to them like this.